File: src/cs4096-software-systems-development/splatoonio/ios/Splatoonio/Splatoonio/Player.m
//
//  Player.m
//  Splatoonio
//
//  Created by Adam Evans on 8/25/17.
//  Copyright © 2017 Kijug Software. All rights reserved.
//

#import "Player.h"


#define PLAYER_ID_NONE			-1
#define PLAYER_START_RADIUS		2.0		// 2 meters seems like a good starting radius (4 meter wide circle)


@implementation Player


@synthesize playerID;
@synthesize teamColor;
@synthesize brushRadius;
@synthesize name;
@synthesize path;


// =================================================================================
#pragma mark -						Initializers
// =================================================================================

- (id)init
{
	self = [super init];
	
	if (self != nil)
	{
		playerID = PLAYER_ID_NONE;
		teamColor = TEAM_RED;
		brushRadius = PLAYER_START_RADIUS;
		name = @"ERROR: No name";
		path = [[NSMutableArray<PathPoint*> alloc] init];
	}
	
	return self;
}

// =================================================================================
// =================================================================================
// =================================================================================




@end

File: src/cs4096-software-systems-development/splatoonio/ios/Splatoonio/Splatoonio/Player.swift
import Foundation
import CoreLocation


class Player
{
	static let INITIAL_BRUSH_RADIUS:Float = 3.0
	
	private var loc:CLLocation?
	
	var id:PlayerID
	var team:TeamColor
	var brushRadius:Float		// in meters
	var distance:Double
	var name:String
	var location:CLLocation?
	{
		get
		{
			return loc;
		}
		set
		{
			if let old = loc, let new = newValue
			{
				distance += new.distance(from:old)
			}
			loc = newValue
		}
	}
	
	
	// =================================================================================
	//									Initializers
	// =================================================================================
	
	init()
	{
		id = 0
		team = .none
		brushRadius = Player.INITIAL_BRUSH_RADIUS
		distance = 0
		loc = nil
		name = ""
	}
	
	// =================================================================================
}
